### PR TITLE
Ask user before killing all running collectd instances

### PR DESCRIPTION
**Pull request by udaysagar2177**

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -292,11 +293,37 @@ main() {
     copy_configs
     verify_configs
 
+    # Check if killall command is available
+    which killall > /dev/null
+    if [ $? -ne 0 ]; then
+        printf "Unable to find killall command, installing it now...\n"
+        printf "killall is required to stop all running collectd instances\n"
```
Comment (mdubbyap) [inline install.sh]: have we not figured out which os we are yet?  Wouldn't we want to do this logic exactly once and set a variable?

File: install.sh [hunk 2]
```diff
@@ -292,11 +293,37 @@ main() {
...
+            ;;
+            *)
+        esac
+    fi
+
+    echo "Stopping currently running collectd"
+    if [ ${USE_SERVICE_COLLECTD} -eq 1 ]; then
+        service collectd stop
+    fi
+    killall collectdmon
```
Comment (mdubbyap) [inline install.sh]: probably want to dev null the outputs so we don't get "collectd: no process found" as the default case for both of these

**Comment by udaysagar2177** [comment]
We don't have any local variable called "hostOS" but we already have a function called "determine_os" to calculate the hostOS value. So, I am using that here.

**Comment by udaysagar2177** [comment]
@mdubbyap I have changed the code to take user decision before killing collectd

**Comment by udaysagar2177** [comment]
@mdubbyap I have updated the code with respect to the comments from @mpetazzoni. I think Max's comments are lost when I force pushed into this branch. You can find those comments here: https://github.com/signalfx/signalfx-collectd-configs/commit/7132d0f4f3870b3b8c773a89896fb581435ccc34

**Comment by udaysagar2177** [comment]
@ramjothikumar I checked the update against centos and amazon as well.

**Pull request merged**